Source/ACE/Entity/Player.cs

```diff
@@ -1570,14 +1570,44 @@ protected override void SendUpdatePosition()
             Session.Network.EnqueueSend(msg);
         }
 
-        private void HandleUnequip(Container container, WorldObject item, uint location)
+        private void HandleUnequip(Container container, WorldObject item, uint location, ActionChain inContainerChain)
         {
             UpdateAppearance(container, item.Guid.Full);
-            Session.Network.EnqueueSend(new GameMessagePutObjectInContainer(Session, container.Guid, item, location),
+            if (item.CurrentWieldedLocation > EquipMask.RingLeft)
+            {
+                // We are coming from a weapon, wand or shield slot.
+                Children.Remove(Children.Find(s => s.Guid == item.Guid.Full));
+                item.Parent = null;
+                // Magic numbers - need to understand and fix.
+                item.ParentLocation = null;
+                item.CurrentWieldedLocation = null;
+                inContainerChain.AddAction(this, () => CurrentLandblock.EnqueueBroadcast(Location,
+                    Landblock.MaxObjectRange, new GameMessageRemoveObject(item)));
+                item.Location = null;
+                item.ContainerId = container.Guid.Full;
+                item.SetPhysicsDescriptionFlag(item);
+            }
+            else
+            {
+                item.Parent = null;
+                item.ParentLocation = null;
+                item.CurrentWieldedLocation = null;
+                item.Location = null;
+                item.ContainerId = container.Guid.Full;
+                item.SetPhysicsDescriptionFlag(item);
+            }
+
+            inContainerChain.AddDelaySeconds(0.001);
```

Comment:
Not sure I understand this delay.  Weird code like this should at-minimum be commented.